### ISSUE TITLE
Gtk4 - example

### ISSUE THREAD

**Issue by d3im** [jaseg/python-mpv#222]
Trying to modify PyGobject example for Gtk4
I can't get over this line - get XID of window since widget lacks window property
```python
self.mpv = mpv.MPV(wid=str(widget.get_property("window").get_xid()))
```

```python
TypeError: object of type `GtkFrame' does not have property `window'
```

**Comment by Maryll-castelino** [comment]
did you find a solution?

**Comment by bscubed** [comment]
I'm also in need of a solution. As of now, I've managed to render in GTK4 using OpenGL with [this gist](https://gist.github.com/jaseg/657e8ecca3267c0d82ec85d40f423caa), but it's very buggy when resizing. Any workarounds for native MPV?

**Comment by Maryll-castelino** [comment]
This worked for me 

`self.mpv = mpv.MPV(wid=str(GdkX11.X11Surface.get_xid(self.get_surface())))`

**Comment by trin94** [comment]
@d3im You are trying to access a X11 window function on GTK 4.0. This will definitely not work on Wayland, too. And GTK 4.0's main focus is Wayland. There are even plans to drop X11 support in GTK 5.0: https://www.phoronix.com/scan.php?page=news_item&px=GTK5-Might-Drop-X11

The render context API should be the way to go. I have experimented with Qt 5 and Qt 6: https://github.com/trin94/qtquick-mpv
Maybe that helps you to get started :smiley: The concepts should be similar for GTK and Qt.

@bscubed Consider using 

```python
self._ctx.render(
    flip_y=True,
    opengl_fbo={'w': width, 'h': height, 'fbo': fbo},
    block_for_target_time=False,
)
```

Documentation is located here: https://github.com/mpv-player/mpv/blob/master/libmpv/render.h#L300-L316=

As stated, you may want to set `video-timing-offset` to `0` additionally.

**Comment by trin94** [comment]
I've created a little poc: https://github.com/trin94/python-mpv-gtk4
Enjoy!

**Comment by trin94** [comment]
@jaseg Yes, the url is stable. You/I can also make a gist out of it so that we can link it in the README :+1: Whatever you chose, we should also make people aware of the thing with the `block_for_target_time=False` and `video-timing-offset=0`: https://github.com/jaseg/python-mpv/issues/222#issuecomment-1179721210. It can then summarize (and possibly replace) `PyGObject embedding` and `Using OpenGL from PyGObject` from the README.

**Comment by trin94** [comment]
Hi @jaseg,

I'd be happy to add the gtk4 example to your project. However, I think there's too much code to put it into the README. What about an `examples` folder in the root of this repository? Just let me know if that's okay with you or if you prefer another solution and I'll prepare a PR 😃

**Comment by trin94** [comment]
I have archived the example repo under https://github.com/trin94/python-mpv-gtk4 with the intention to remove it in a couple of weeks. As a replacement, I've created a gist: https://gist.github.com/trin94/3381395adc8b2c3fea81a38b9a385369

@jaseg 
Feel free to either copy and adapt or link it directly in the README 😃
